### ISSUE TITLE
Compare Classes/Jar

### ISSUE THREAD

**Issue by escequi** [Col-E/Recaf#643]
Hello, i did some reverse on this Jar but i swear i don't remember what i changed (was on a 12 hour analysis so yea hard to remember it all and i didn't really take notes), is there any way i can compare all the classes in the jar with another jar (the unmodified version)? (bytecode)

**Comment by Col-E** [comment]
If you check out the WIP 3x releases this feature sort-of exists:

https://user-images.githubusercontent.com/21371686/233784793-3df846f1-7dcd-46b0-b9c3-4a7e99dbf725.mp4

**Comment by escequi** [comment]
> If you check out the WIP 3x releases this feature sort-of exists:
> java_Uur5ktzyJZ.mp4

hi, thanks for the reply; from what i understand, this only work if you haven't closed the workspace yet?

**Comment by Col-E** [comment]
This is correct. Having a more comprehensive jar comparison scheme is planned, but not yet implemented yet.

**Comment by escequi** [comment]
> This is correct. Having a more comprehensive jar comparison scheme is planned, but not yet implemented yet.

Aight, thanks. Is there any way to export all the classes' bytecode to a .txt or something? So i can use notepad++ to compare...

**Comment by Col-E** [comment]
Not currently, but a disassembly based text comparison strategy is what is roughly planned for the future.

3X has a WIP class-level assembler. You can go that route for now, or do basic decomp diffs.

**Comment by NebelNidas** [comment]
You can use https://github.com/sfPlayer1/Matcher for comparing JARs

**Comment by eleses** [comment]
IntelliJ idea can compare jars quite nicely too, even in the community edition.

**Comment by Col-E** [comment]
#676 - Similar in idea, allows mapping based on structural similarity. A proper diff view still pending.
